src/transport.rs

```diff
@@ -117,7 +117,12 @@ impl WireNetwork {
                 match Message::unmarshal_binary(&mut &message[..]) {
                     Ok(deser) => {
                         debug!("> Received raw message {}", deser.type_byte());
-                        let to_process = decoder.decode(deser);
+
+                        let to_process =  decoder.decode(deser).map_err(|e|{
+                            error!("Unable to process the message through the decoder: {}",e);
+                        })
+                        .ok().flatten();
+
                         if let Some(message) = to_process {
```

Comment:
Since you're doing already a pattern here I would suggest to avoid `map_err` and just have a `match to_process`.